arrow-integrations-jackson-module/src/main/kotlin/arrow/integrations/jackson/module/internal/ProjectField.kt

```diff
@@ -0,0 +1,5 @@
+package arrow.integrations.jackson.module.internal
+
+import arrow.core.Option
+
+class ProjectField<T>(val fieldName: String, val getOption: (T) -> Option<*>)
```

Comment:
Nit: Should this be a static nested class of `UnionTypeDeserializer ` to give it better namespacing? I had to look around a bit to see where this belonged to and why.